server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/DefaultJobProcessUpdateListener.java

```diff
@@ -58,44 +51,12 @@ public void onEvent(DefaultJobProcessUpdateEvent event) {
         TaskResult taskResult = event.getTaskResult();
         JobIdentity identity = taskResult.getJobIdentity();
         JobEntity jobEntity = stdTaskFrameworkService.find(identity.getId());
-
-        Optional<ScheduleTask> scheduleTaskEntityOptional = scheduleTaskService.findByJobId(jobEntity.getId());
-        if (scheduleTaskEntityOptional.isPresent()) {
-            updateScheduleTask(taskResult, scheduleTaskEntityOptional.get());
-            return;
-        }
-        Optional<TaskEntity> taskEntityOptional = taskService.findByJobId(jobEntity.getId());
-        taskEntityOptional.ifPresent(taskEntity -> updateTask(taskResult, taskEntity));
+        scheduleTaskService.findByJobId(jobEntity.getId())
+                .ifPresent(taskEntity -> updateScheduleTask(taskEntity));
     }
 
-    private void updateScheduleTask(TaskResult taskResult, ScheduleTask taskEntity) {
-        taskEntity.setProgressPercentage(taskResult.getProgress());
-        taskEntity.setStatus(taskResult.getStatus().convertTaskStatus());
-        taskEntity.setResultJson(taskResult.getResultJson());
-        if (taskResult.getExecutorEndpoint() != null) {
-            UriComponents uc = UrlUtils.getUriComponents(taskResult.getExecutorEndpoint());
-            ExecutorInfo executorInfo = new ExecutorInfo();
-            executorInfo.setHost(uc.getHost());
-            executorInfo.setPort(uc.getPort());
-            taskEntity.setExecutor(JsonUtils.toJson(executorInfo));
-        }
-        scheduleTaskService.update(taskEntity);
+    private void updateScheduleTask(ScheduleTask taskEntity) {
+        scheduleTaskService.updateStatusById(taskEntity.getId(), taskEntity.getStatus());
```

Comment:
change this method name to updateScheduleTaskStatus and the following debug log print new status

Reply:
done